examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,274 @@
+ 
+
+def plot(xlabel, ylabel, values, inertiae):
+    arr = np.asarray(inertiae)
+    for i in range(arr.shape[1]):
+        plt.scatter(values, arr[:, i], alpha=.2, c='gray')
+    plt.plot(values, np.mean(arr, axis=1), c='r')
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.show()
+
+
+##############################################################################
+# Number of fixed cluster centers
+# ===============================
+#
+# In this experiment, we generate data from 10 clusters and see how the number
+# of fixed cluster influence the intertia.
+
+study_value = []
+study_inert = []
+
+for i in range(10):
+    study_value.append(i)
+    inert = evaluate(5000, 100, 20, 10, 10, i, 0.01, None)
+    study_inert.append(inert)
+    
+plot('Number of fixed center clusters /10', 'Mean inertia over 100 runs',
+     study_value, study_inert)
+
+##############################################################################
+# It may seem surprising that the
+# intertia is proportional to the number of fixed clusters, which means that
+# the intertia of the original centers used to generate the data is higher
+# than the centers after KMeans, but it is logical since KMeans optimizes for
+# it.
+# 
+# # Reassignment ratio
+# ==================
+#
+# We expect the fixed clusters to "get in the way" of clusters trying to move
+# around to reach their cluster. Fortunately, the KMeans++ initialization
+# usually takes care of this by positioning initialization centers in the best
+# areas.
+
+
+study_value = []
+study_inert = []
+
+for i in [0., 0.01, 0.05, 0.1, 0.15, 0.2, 0.3, 0.5]:
+    study_value.append(i)
+    inert = evaluate(5000, 100, 20, 10, 10, 5, i, None)
+    study_inert.append(inert)
+    
+plot('Reassignment ratio', 'Mean inertia over 100 runs',
+     study_value, study_inert)
+
+##############################################################################
+# In the end, we observe that the resassignment ratio has absolutely no impact
+# on the intertia which is reassuring.
+# 
+# # Recenter every
+# ==============
+#
+# The effect of this parameter has already been described above. In its
+# highest value, this parameter leads to the following behavior: it initialize
+# centers on the fixed center positions, let the KMeans run, and in the end
+# move the centers closest to the fixed position on the fixed spots.
+
+study_value = []
+study_inert = []
+
+for i in [None, 1, 2, 3, 5, 10, 20, 50, 100]:
+    study_value.append(i if i is not None else 0)
+    inert = evaluate(5000, 100, 20, 10, 10, 5, 0.1, i)
+    study_inert.append(inert)
+    
+plot('Recenter every n iterations', 'Mean inertia over 100 runs', study_value, study_inert)
+
+
+##############################################################################
+# Surprisingly, this setting seems to give the best result. This result is a
+# bit more surprising but a better inertia does not necessarily means a result
+# closer to the data.
+#
+# Active learning like-experiment
```

Comment:
Shouldn't we also give the usual active learning plot of accuracy over the iterations ?

Reply:
The thing is that I'm not doing an real active learning experiment, because I wanted to avoid having to introduce Zhdanov in a simple example. In this one, I just explore 10 different subset of the data to model the fact that active learning does not consider the whole dataset at each iteration. I will find a better example.

Reply:
Fixed in an upcoming commit.